web/src/hotspots.ts

```diff
@@ -57,27 +60,34 @@ const HOTSPOT_LOCATIONS = new Map([
     ],
 ]);
 
-const meta = {
+const meta: {
+    opened_hotspot_name: null | string;
+} = {
     opened_hotspot_name: null,
 };
 
-function compute_placement($elt, popover_height, popover_width, prefer_vertical_positioning) {
-    const client_rect = $elt.get(0).getBoundingClientRect();
+function compute_placement(
+    $elt: JQuery,
+    popover_height: number,
+    popover_width: number,
+    prefer_vertical_positioning: boolean,
+): Position {
+    const client_rect = $elt.get(0)!.getBoundingClientRect();
     const distance_from_top = client_rect.top;
     const distance_from_bottom = message_viewport.height() - client_rect.bottom;
     const distance_from_left = client_rect.left;
     const distance_from_right = message_viewport.width() - client_rect.right;
 
     const elt_will_fit_horizontally =
-        distance_from_left + $elt.width() / 2 > popover_width / 2 &&
-        distance_from_right + $elt.width() / 2 > popover_width / 2;
+        distance_from_left + ($elt.width() ?? 0) / 2 > popover_width / 2 &&
+        distance_from_right + ($elt.width() ?? 0) / 2 > popover_width / 2;
```

Comment:
I think you can probably avoid needing to add these operators, which suggest a `0` would not be an error, by adding some `assert` statements above about `$elt`.